notebooks/src/el-paso.ipynb

```diff
@@ -186,21 +190,47 @@
     "  if bw is not None:\n",
     "      q.show_prediction(samples=q_samples, show_community=True, percent_kept=0.9, bw=bw)\n",
     "  else:\n",
-    "      q.show_prediction(samples=q_samples, show_community=True, percent_kept=0.9)"
+    "      q.show_prediction(samples=q_samples, show_community=True, percent_kept=0.9)\n",
+    "\n",
+    "\n",
+    "def sample_from_ensemble(models, params, weights=None, fallback=False, default=None):\n",
+    "    \"\"\"Sample models in proportion to weights and execute with\n",
+    "    model_params. If fallback is true then call different model from\n",
+    "    ensemble if the selected model throws an error. If Default is not\n",
+    "    None then return default if all models fail\n",
+    "\n",
+    "    \"\"\"\n",
+    "\n",
+    "    model = ergo.random_choice(models, weights)\n",
+    "    try:\n",
+    "        result = model(**params)\n",
+    "        if np.isnan(result):\n",
+    "            raise KeyError\n",
+    "        return result\n",
+    "    except:\n",
```

Comment:
Catch a more specific exception

Reply:
My thinking is that we would want to catch all-cause model failure and then call the fallback models. Given that we won't know what errors the model may throw (outside the scope of the function) I have the omnibus catch. Happy to just catch the KeyError for now (we wrote the models and know that is enough) but then we should revisit this if this becomes a longer-lived function used across notebooks

Reply:
(The worry with catch-all is that there could be unexpected bugs that we don't notice.)